functests/utils/consts.go

```diff
@@ -25,6 +32,18 @@ func init() {
 	}
 
 	NodesSelector = os.Getenv("NODES_SELECTOR")
+
+	NodeSelectorLabels = map[string]string{
+		fmt.Sprintf("%s/%s", LabelRole, RoleWorkerCNF): "",
+	}
+
+	if discovery.Enabled() {
+		profile, err := discovery.GetDiscoveryPerformanceProfile()
+		if err == nil {
+			PerformanceProfileName = profile.Name
+		}
+		NodeSelectorLabels = profile.Spec.NodeSelector
+	}
```

Comment:
I think you should use the content of `NODES_SELECTOR` (if present) and add another entry to the map here.